src/PrizePool.ts

```diff
@@ -331,9 +344,10 @@ export class PrizePool {
 
   /**
    * Fetches the addresses to build an instance of an ethers Contract for the underlying Token
+   * @param options optional settings or toggles
    * @returns an ethers contract for the underlying token
    */
-  async getTokenContract(): Promise<Contract> {
+  async getTokenContract(options?: { eip2612?: boolean }): Promise<Contract> {
     if (this.tokenContract !== undefined) return this.tokenContract
```

Comment:
If `tokenContract` gets initialized it'll short circuit here. If it's initialized with an ERC20 ABI not a ERC20AbiWithPermit ABI, when `getPermitAndDepositSignaturePromise` runs and uses the `tokenContract` the contract won't have the permit methods. Probably better to have a separate `getEip2612TokenContract` function that calls `getTokenContract` and returns a new ethers `Contract`.

Reply:
As of a7d55c7, the `tokenContract` is now cleared from the `PrizePool` object when this function is called with the optional `eip2612` variable.